### PR TITLE
Added edit and remove meme commands.

### PR DESCRIPTION
**Pull request by naps250**
Added edit meme command.
Added remove meme command.
Refactoring.

(cherry picked from commit f50cc1094ab15dfd4cf4903e0d059c2b5a0f6a57)

### CODE REVIEW

File: SteakBot.Core/Modules/MemeModule.cs [hunk 1]
```diff
@@ -42,10 +42,60 @@ public async Task AddMeme(string name, string value, string description)
             }
             else
             {
-                replyMessage = "Saving failed! ;(";
+                replyMessage = "Add failed! ;(";
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: "Adding" ?

File: SteakBot.Core/Modules/MemeModule.cs [hunk 2]
```diff
@@ -42,10 +42,60 @@ public async Task AddMeme(string name, string value, string description)
...
+                {
+                    replyMessage = newValue;
+                }
+            }
+            else
+            {
+                replyMessage = "Edit failed! ;(";
+            }
+
+            await ReplyAsync(replyMessage, embed: embed?.Build());
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: This is a straight copy of `AddMeme()`. Please pull as much as possible into a reusable method.

File: SteakBot.Core/Modules/MemeModule.cs [hunk 3]
```diff
@@ -42,10 +42,60 @@ public async Task AddMeme(string name, string value, string description)
...
+            if (result)
+            {
+                replyMessage = "Removed";
+            }
+            else
+            {
+                replyMessage = "Remove failed! ;(";
+            }
+
+            await ReplyAsync(replyMessage);
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: This can be condensed to a single line using the ternary operator.

Comment (penev92) [reply SteakBot.Core/Modules/MemeModule.cs]: In fact the whole method body can be condensed to 1 line, or 2 at most.

**Review by penev92 — CHANGES_REQUESTED** [review]

File: SteakBot.Core/Modules/MemeModule.cs (hunk 1 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Yes, because you're invoking addMeme.

File: SteakBot.Core/Modules/MemeModule.cs (hunk 3 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: I will convert the if to use the ternary operator but I am not condensing everything into a single line!

File: SteakBot.Core/Modules/MemeModule.cs (hunk 2 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Corrected.

File: SteakBot.Core/Modules/MemeModule.cs [hunk 4]
```diff
@@ -20,20 +20,50 @@ public MemeModule(MemeService memeService)
         public async Task AddMeme(string name, string value, string description)
         {
             var replyMessage = string.Empty;
-            EmbedBuilder embed = null;
+            EmbedBuilder embed = new EmbedBuilder();
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: Both local fields are now unused and obsolete.

File: SteakBot.Core/Modules/MemeModule.cs [hunk 5]
```diff
@@ -20,20 +20,50 @@ public MemeModule(MemeService memeService)
...
 
-            if (result)
+            await SendReply(result, type, value, "Add failed! ;(");
+        }
+
+        [Command("editMeme")]
+        public async Task EditMeme(string oldName, string newName, string newValue, string newDescription)
+        {
+            var replyMessage = string.Empty;
+            EmbedBuilder embed = new EmbedBuilder();
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: Both local fields are now unused and obsolete.

File: SteakBot.Core/Modules/MemeModule.cs [hunk 6]
```diff
@@ -20,20 +20,50 @@ public MemeModule(MemeService memeService)
...
+
+            await ReplyAsync(replyMessage);
+        }
+
+        #region Private methods
+
+        private Task SendReply(bool commandResult, MemeResultType memeType, string value, string failureMessage)
+        {
+            var replyMessage = string.Empty;
+            EmbedBuilder embed = new EmbedBuilder();
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: Please use `var` whenever possible.

File: SteakBot.Core/Modules/MemeModule.cs [hunk 7]
```diff
@@ -20,20 +20,50 @@ public MemeModule(MemeService memeService)
...
+            var result = _memeService.RemoveCommand(name);
+
+            var replyMessage = result ? "Removed" : "Remove failed! ;(";
+
+            await ReplyAsync(replyMessage);
+        }
+
+        #region Private methods
+
+        private Task SendReply(bool commandResult, MemeResultType memeType, string value, string failureMessage)
```
Comment (penev92) [inline SteakBot.Core/Modules/MemeModule.cs]: Minor nit: the name here sounds like the library functions.

**Review by penev92 — COMMENTED** [review]
Left a few comments for slight adjustments. Otherwise looks good now.

File: SteakBot.Core/Modules/MemeModule.cs (hunk 6 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Corrected.

File: SteakBot.Core/Modules/MemeModule.cs (hunk 7 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Renamed.

File: SteakBot.Core/Modules/MemeModule.cs (hunk 5 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Corrected.

File: SteakBot.Core/Modules/MemeModule.cs (hunk 4 above)
Comment (naps250) [reply SteakBot.Core/Modules/MemeModule.cs]: Corrected.

**Review by penev92 — APPROVED** [review]
Looks good now 👍

**Pull request merged**